addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1351,6 +1353,212 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
     );
 });
 
+QUnit.test('chat window: scroll position conservation on folded', async function (assert) {
+    assert.expect(3);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                is_minimized: false,
+                is_pinned: true,
+                name: "General",
+                state: 'open',
+                uuid: 'channel-20-uuid',
+            }],
+        },
+    });
+    this.data['mail.message'].records = [{
+        author_id: [7, "Demo"],
+        body: "<p>test</p>",
+        channel_ids: [20],
+        id: 100,
+        message_type: 'comment',
+        model: 'mail.channel',
+        res_id: 20,
+    }];
+    [...Array(20).keys()].map(i => {
+        this.data['mail.message'].records.push({
+            author_id: [7, "Demo"],
+            body: "<p>body</p>",
+            channel_ids: [20],
+            date: "2019-04-20 10:00:00",
+            id: i + 10,
+            message_type: 'comment',
+            model: 'mail.channel',
+            record_name: 'General',
+            res_id: i + 1,
+        });
+    });
+
+    await this.start();
+    await afterNextRender(() => document.querySelector(`.o_MessagingMenu_toggler`).click());
+    await afterNextRender(() =>
+        document.querySelector(`.o_MessagingMenu_dropdownMenu .o_NotificationList_preview`).click()
+    );
+    // Set a scroll position to chat window
+    document.querySelector(`.o_ThreadViewer_messageList`).scrollTop = 142;
+    assert.strictEqual(
+        document.querySelector(`.o_ThreadViewer_messageList`).scrollTop,
+        142,
+        "verify chat window initial scrollTop"
+    );
+
+    // fold chat window
+    await afterNextRender(() => document.querySelector('.o_ChatWindow_header').click());
+    assert.containsNone(
+        document.body,
+        ".o_ThreadViewer",
+        "chat window should be folded so no thread viewer should be present"
+    );
+
+    // unfold chat window
+    await afterNextRender(() => document.querySelector('.o_ChatWindow_header').click());
+    assert.strictEqual(
+        document.querySelector(`.o_ThreadViewer_messageList`).scrollTop,
+        142,
+        "chat window scrollTop should still be the same when chat window is unfolded"
+    );
+});
+
+QUnit.test('chat window: state conservation on toggle home menu when folded', async function (assert) {
```

Comment:
It's the same as "chat window: state conservation on toggle home menu" just with a folding of the chatwindow before going on/out of home menu. This could be exploded in multiple tests but those tests are going to be really really identicals in the steps we do so it won't make it that more simple I think, there will just be less assertions

Reply:
Or you could split show home menu / hide home menu maybe

Reply:
I don't mind lots of tests which reuse same steps but don't have intermediate assertions.
It's better for a test to focus on less assertions.
All of these tests should obviously be tagged `[technical]` for the same reason as `chat window: state conservation on toggle home menu`.